### DISCUSSION TITLE
need help: DTU has problems in production operations

### DISCUSSION THREAD

**Discussion by 2000ede** [tbnobody/OpenDTU#1754] [Q&A] [answered]
Hello,
My DTU runs unstable (sometimes it works fine for hours, then it freezes briefly, then completely, sometimes it just sets the limit to 0 and stays there).

Can you see anything from the log? (I am a Beginner).

The log was created when the DTU froze again. I can still access the DTU and the headings are there, but in the live view everything is white.

I have connected 2 HMS inverters. Interval to 5 seconds.

I request from outside via API every 4 seconds. I also send a limit every 4 seconds, but only if necessary.

The DTU is spatially accurate to the inverters. Reception should actually be very good.

Am I overwhelming the system with the requests?

What can you possibly see from the log?

Thanks for help!

`Websocket: [/livedata][22] disconnect
10:52:24.349 > Alloc failed: sendDataTaskCb, 74
10:52:25.465 > Call to /api/livedata/status temporarely out of resources. Reason: "std::bad_alloc".
10:52:26.347 > Alloc failed: sendDataTaskCb, 74
10:52:27.307 > Fetch inverter: 116491436565
10:52:27.307 > Request SystemConfigPara
10:52:27.526 > TX RealTimeRunData 865.00 MHz --> 15 91 43 65 65 80 16 98 04 80 0B 00 65 CF 30 5A 00 00 00 00 00 00 00 00 0D 63 E8 
10:52:27.739 > Alloc failed: sendDataTaskCb, 74
10:52:27.930 > Interrupt received
10:52:28.138 > RX 865.00 MHz --> 95 91 43 65 65 80 16 98 04 01 00 01 01 3F 01 3F 00 70 00 6F 01 65 01 63 00 00 54 | -10 dBm
10:52:28.339 > Interrupt received
10:52:28.553 > RX 865.00 MHz --> 95 91 43 65 65 80 16 98 04 02 0D 5E 00 00 0D 2E 00 37 00 36 01 3F 01 41 00 6F 2F | -10 dBm
10:52:28.758 > Interrupt received
10:52:28.958 > RX 865.00 MHz --> 95 91 43 65 65 80 16 98 04 03 00 70 01 63 01 69 00 00 0D 47 00 00 0D AC 00 36 E9 | -10 dBm
10:52:29.159 > Interrupt received
10:52:29.369 > RX 865.00 MHz --> 95 91 43 65 65 80 16 98 04 04 00 37 09 14 13 8A 05 4E 00 01 00 3A 03 E7 00 9A F4 | -10 dBm
10:52:29.567 > Interrupt received
10:52:29.675 > RX 865.00 MHz --> 95 91 43 65 65 80 16 98 04 85 00 05 42 C2 4D | -10 dBm
10:52:29.767 > RX Period End
10:52:29.767 > Success
10:52:29.876 > TX SystemConfigPara 865.00 MHz --> 15 91 43 65 65 80 16 98 04 80 05 00 65 CF 30 5A 00 00 00 00 00 00 00 00 C3 6C 27 
10:52:30.083 > Interrupt received
10:52:30.287 > RX 865.00 MHz --> 95 91 43 65 65 80 16 98 04 81 00 01 03 E8 00 00 03 E8 00 00 00 00 00 00 3C F8 09 | -10 dBm
10:52:30.489 > RX Period End
10:52:30.489 > Success
10:52:30.698 > Alloc failed: sendDataTaskCb, 74
10:52:30.898 > Alloc failed: sendDataTaskCb, 74
10:52:31.106 > Alloc failed: sendDataTaskCb, 74
10:52:31.309 > Fetch inverter: 116491958656
10:52:31.516 > TX RealTimeRunData 865.00 MHz --> 15 91 95 86 56 80 16 98 04 80 0B 00 65 CF 30 5E 00 00 00 00 00 00 00 00 CD 51 18 
10:52:31.722 > Call to /api/livedata/status temporarely out of resources. Reason: "std::bad_alloc".
10:52:31.722 > Interrupt received
10:52:31.925 > RX 865.00 MHz --> 95 91 95 86 56 80 16 98 04 01 00 01 01 41 01 40 00 6F 00 6F 01 66 01 65 00 00 49 | -43 dBm
10:52:32.127 > Interrupt received
10:52:32.366 > RX 865.00 MHz --> 95 91 95 86 56 80 16 98 04 04 00 36 09 02 13 8A 05 53 00 00 00 3B 03 E8 00 95 F8 | -43 dBm
10:52:32.474 > Interrupt received
10:52:32.645 > RX 865.00 MHz --> 95 91 95 86 56 80 16 98 04 85 00 05 27 74 98 | -43 dBm
10:52:32.848 > RX Period End
10:52:32.848 > Middle missing
10:52:32.848 > Request retransmit: 2
10:52:32.848 > TX RequestFrame 865.00 MHz --> 15 91 95 86 56 80 16 98 04 82 49 
10:52:33.050 > Interrupt received
10:52:33.280 > RX 865.00 MHz --> 95 91 95 86 56 80 16 98 04 02 3E 2D 00 00 3D B5 00 36 00 36 01 40 01 40 00 70 A2 | -43 dBm
10:52:33.459 > RX Period End
10:52:33.459 > Middle missing
10:52:33.459 > Request retransmit: 3
10:52:33.459 > TX RequestFrame 865.00 MHz --> 15 91 95 86 56 80 16 98 04 83 48 
10:52:33.673 > Interrupt received
10:52:33.787 > RX 865.00 MHz --> 95 91 95 86 56 80 16 98 04 03 00 6F 01 68 01 66 00 00 3F B6 00 00 3D CD 00 36 66 | -43 dBm
10:52:33.974 > RX Period End
10:52:33.974 > Success
10:52:34.176 > Alloc failed: sendDataTaskCb, 74
10:52:34.284 > Alloc failed: sendDataTaskCb, 74
10:52:34.501 > Alloc failed: sendDataTaskCb, 74
10:52:35.314 > Fetch inverter: 116491436565
10:52:35.456 > TX RealTimeRunData 865.00 MHz --> 15 91 43 65 65 80 16 98 04 80 0B 00 65 CF 30 62 00 00 00 00 00 00 00 00 CC 50 22 
10:52:35.675 > Alloc failed: sendDataTaskCb, 74
10:52:39.364 > Interrupt received
10:52:45.349 > Alloc failed: sendDataTaskCb, 74
10:52:46.425 > Call to /api/livedata/status temporarely out of resources. Reason: "std::bad_alloc".
10:52:47.327 > Fetch inverter: 116491958656
10:52:47.427 > TX RealTimeRunData 865.00 MHz --> 15 91 95 86 56 80 16 98 04 80 0B 00 65 CF 30 6E 00 00 00 00 00 00 00 00 CC 05 7D 
10:52:47.628 > Alloc failed: sendDataTaskCb, 74
10:52:47.801 > Interrupt received
10:52:48.011 > RX 865.00 MHz --> 95 91 95 86 56 80 16 98 04 01 00 01 01 41 01 40 00 6F 00 6F 01 65 01 64 00 00 4B | -43 dBm
10:52:48.214 > Interrupt received
10:52:48.483 > RX 865.00 MHz --> 95 91 95 86 56 80 16 98 04 02 3E 2D 00 00 3D B5 00 36 00 36 01 3F 01 40 00 70 DD | -43 dBm
10:52:48.626 > Interrupt received
10:52:48.839 > RX 865.00 MHz --> 95 91 95 86 56 80 16 98 04 03 00 6F 01 67 01 64 00 00 3F B6 00 00 3D CD 00 36 6B | -43 dBm
10:52:49.033 > Interrupt received
10:52:49.232 > RX 865.00 MHz --> 95 91 95 86 56 80 16 98 04 04 00 36 09 07 13 8A 05 4E 00 01 00 3A 03 E7 00 95 EF | -43 dBm
10:52:49.432 > Interrupt received
10:52:49.641 > RX 865.00 MHz --> 95 91 95 86 56 80 16 98 04 85 00 05 13 BE 66 | -43 dBm
10:52:49.841 > RX Period End
10:52:49.841 > Success
10:52:50.060 > Alloc failed: sendDataTaskCb, 74
10:52:50.253 > Alloc failed: sendDataTaskCb, 74
10:52:50.466 > Alloc failed: sendDataTaskCb, 74
10:52:51.329 > Fetch inverter: 116491436565
10:52:51.494 > TX RealTimeRunData 865.00 MHz --> 15 91 43 65 65 80 16 98 04 80 0B 00 65 CF 30 72 00 00 00 00 00 00 00 00 0C 9D 3F 
10:52:51.722 > Alloc failed: sendDataTaskCb, 74
10:52:51.821 > Interrupt received
10:52:51.999 > RX 865.00 MHz --> 95 91 43 65 65 80 16 98 04 01 00 01 01 3E 01 3F 00 6F 00 6E 01 63 01 61 00 00 4F | -10 dBm
10:52:52.198 > Interrupt received
10:52:52.312 > RX 865.00 MHz --> 95 91 43 65 65 80 16 98 04 02 0D 5E 00 00 0D 2E 00 37 00 36 01 3F 01 41 00 6E 2E | -10 dBm
10:52:52.439 > Interrupt received
10:52:52.611 > RX 865.00 MHz --> 95 91 43 65 65 80 16 98 04 03 00 6F 01 61 01 67 00 00 0D 47 00 00 0D AD 00 36 FB | -10 dBm
10:52:52.721 > Interrupt received
10:52:52.919 > RX 865.00 MHz --> 95 91 43 65 65 80 16 98 04 04 00 38 09 17 13 8A 05 46 00 01 00 3A 03 E7 00 99 F3 | -10 dBm
10:52:53.119 > Interrupt received
10:52:53.224 > RX 865.00 MHz --> 95 91 43 65 65 80 16 98 04 85 00 05 06 D5 1E | -10 dBm
10:52:53.432 > RX Period End
10:52:53.432 > Success
10:52:53.538 > Call to /api/livedata/status temporarely out of resources. Reason: "std::bad_alloc".
10:52:53.743 > Alloc failed: sendDataTaskCb, 74

![grafik](https://github.com/tbnobody/OpenDTU/assets/37264959/862e0ea3-3c24-498a-8ef2-a64884a454b7)

![grafik](https://github.com/tbnobody/OpenDTU/assets/37264959/4d59e998-6a4a-4641-9c16-88615db040c6)

**Comment by ms49434** [comment, marked as answer]
It looks like there's a memory allocation issue. Please upgrade to the latest OpenDTU release. The Web-API has changed significantly  and its footprint is much smaller now. Unfortunately, applications querying the Web-API (/api/livedata/status) must be upgraded.

**Reply by 2000ede** [reply]
Thanks! It is currently running more stable after the update. But I still have to keep an eye on it.
Since the new version, the log of an WR now always says “time calibration”. Does it indicate another problem?

![Uploading grafik.png…]()
